### PR TITLE
Disabled spiking locked item in multiselect

### PR DESCRIPTION
**Pull request by Aca-jov**
SD-2959, SD-3563

### CODE REVIEW

File: client/app/scripts/superdesk-monitoring/monitoring.js [hunk 1]
```diff
@@ -73,11 +73,9 @@
             if (card.type === 'search' && card.search && card.search.filter.query) {
                 params = card.search.filter.query;
                 if (card.query) {
-                    if (card.search.filter.query.q) {
-                        params.q = '(' + card.query + ') ' + card.search.filter.query.q;
```
Comment (Aca-jov) [inline client/app/scripts/superdesk-monitoring/monitoring.js]: @akintolga as I saw you added this code couple months ago, can you please review this changes that I made, are they ok. I didn't understand why on every search, query is added every time o previous query (even when i delete search), maybe it is needed somewhere else, and it shouldn't be deleted.

File: client/app/scripts/superdesk-monitoring/monitoring.js [hunk 2]
```diff
@@ -73,11 +73,9 @@
...
-                    } else {
-                        params.q = '(' + card.query + ') ';
-                    }
```
Comment (Aca-jov) [inline client/app/scripts/superdesk-monitoring/monitoring.js]: @akintolga as I saw you added this code couple months ago, can you please review this changes that I made, are they ok. I didn't understand why on every search, query is added every time o previous query (even when i delete search), maybe it is needed somewhere else, and it shouldn't be deleted.

File: client/app/scripts/superdesk-monitoring/aggregate.js [hunk 3]
```diff
@@ -335,19 +335,6 @@
             });
         };
 
-        /**
```
Comment (ioanpocol) [inline client/app/scripts/superdesk-monitoring/aggregate.js]: This code reset the current search for every group, you have to keep it.

File: client/app/scripts/superdesk-monitoring/monitoring.js (hunk 2 above)
Comment (ioanpocol) [inline client/app/scripts/superdesk-monitoring/monitoring.js]: You have to keep card.search.filter.query.q, because it is from saved search groups type.

**Comment by ioanpocol** [comment]
Disabled spiking locked item in multiselect: I think that this condition should be added where the multi spike command is implemented:
https://github.com/superdesk/superdesk/blob/master/client/app/scripts/superdesk-archive/module.js#L105

File: client/app/scripts/superdesk-monitoring/aggregate.js (hunk 3 above)
Comment (Aca-jov) [inline client/app/scripts/superdesk-monitoring/aggregate.js]: No, I checked, this function is never called, take a look. there is function:
```     this.search = function(query) {
            this.searchQuery = query;
            _.each(this.groups, function(item) {
                item.query = query;
            });
            _.each(this.spikeGroups, function(item) {
                item.query = query;
            });
        };```
On reset it is called search(null), so no need for this.

Comment (ioanpocol) [inline client/app/scripts/superdesk-monitoring/aggregate.js]: ok

File: client/app/scripts/superdesk-archive/module.js [hunk 4]
```diff
@@ -80,19 +80,21 @@ define([
...
-                    }
-                    return item;
-                }, function(response) {
-                    item.error = response;
-                    if (angular.isDefined(response.data._issues) &&
-                        angular.isDefined(response.data._issues['validator exception'])) {
-                        notify.error(gettext(response.data._issues['validator exception']));
-                    }
-                });
+            if (!item.lock_user) {
```
Comment (sivakuna-aap) [inline client/app/scripts/superdesk-archive/module.js]: There should be a message to the user why the item couldn't be spiked.

Comment (ioanpocol) [inline client/app/scripts/superdesk-archive/module.js]: I think that you have to check if the list of selected items can be spiked and enable multi spike/unspike button if the operation is possible.

**Comment by Aca-jov** [comment]
If one of items is locked, spike button is hidden.

**Comment by ioanpocol** [comment]
:+1:

**Comment by pavlovicnemanja** [comment]
:+1:

**Pull request merged**